Azure/sonic-mgmt
# [BGP] Implement new testcase test_traffic_shift

### Description of PR


Summary:
Fixes # (issue)
Implement a new testcase ```test_traffic_shift```.

- case 1 test_TSA
Verify all routes are announced to bgp monitor, and only loopback routes are announced to neighs.

1. Start exabgp on ptf, and establise BGP session with DUT
2. Setup BGPMON over the BGP session in step 1
3.  Issue ```TSA``` on DUT, and wait some time (say 30 secs)
4. Verify all routes are announced to BGPMON on ptf
5. Verify only loopback routes with certain community are announced to bgp neighbors 

- case 2 test_TSB
Verify all routes are announced to bgp monitor and all routes are announced to neighbors.
1. Start exabgp on ptf, and establise BGP session with DUT
2. Setup BGPMON over the BGP session in step 1
3.  Issue ```TSB``` on DUT, and wait some time (say 30 secs)
4. Verify all routes are announced to BGPMON on ptf
5. Verify all routes are announced to bgp neighbors

Please merge after https://github.com/Azure/sonic-buildimage/pull/5906 and https://github.com/Azure/sonic-buildimage/pull/5968 are merged.

### Type of change




### Approach
#### What is the motivation for this PR?
This PR is to implement new testcase ```test_traffic_shift```.

#### How did you do it?
1. Use exabgp to dump all received routes on BGPMON, and parsed into a set;
2. Retrieve all routes from ceos and parsed into a dict;
3. Retrieve all routes from dut and parsed into a dict;
4. Compare routes retrieved in step 3 with that retrieved in step 1 and step 2.

#### How did you verify/test it?
Verified on vtestbed t1
```
py.test --inventory veos_vtb --host-pattern vlab-01 --module-path ../ansible/library/ --testbed vms-kvm-t1 --testbed_file vtestbed.csv --junit-xml=tr.xml --log-cli-level warn -k 'not test_restart_syncd' --topology=t1,t0,any,util bgp/test_traffic_shift.py
========================================================================================= test session starts =========================================================================================
collected 2 items                                                                                                                                                                                     

bgp/test_traffic_shift.py::test_TSA ^@PASSED                                                                                                                                                      [ 50%]
bgp/test_traffic_shift.py::test_TSB ^@PASSED                                                                                                                                                      [100%]

-------------------------------------------------------------------------- generated xml file: /data/sonic-mgmt/tests/tr.xml --------------------------------------------------------------------------
===================================================================================== 2 passed in 822.05 seconds ======================================================================================
```

#### Any platform specific information?
No.

#### Supported testbed topology if it's a new test case?
Only run on T1 topo.

### Documentation
